sim-rs/src/sim.rs

```diff
@@ -0,0 +1,90 @@
+use std::time::Duration;
+
+use rand::{thread_rng, Rng};
+use rand_chacha::{rand_core::SeedableRng, ChaChaRng};
+use tokio::time;
+
+use crate::{
+    config::{PoolId, SimConfiguration},
+    events::EventTracker,
+};
+
+pub struct Simulation {
+    pools: Vec<Pool>,
+    vrf_max_score: u64,
+    current_slot: u64,
+}
+
+impl Simulation {
+    pub fn new(config: SimConfiguration) -> Self {
+        let mut rng = ChaChaRng::from_rng(thread_rng()).expect("couldn't initialize RNG");
```

Comment:
probably want the seed as part of the SimConfiguration

Reply:
@Quantumplation I don't want to make guarantees of determinism unless it's necessary. The `ce_netsim` library forces everything to be "realtime", and I'm worried that determinism will be hard to maintain if actual threads are interacting.

Reply:
Even outside of making it deterministic, just generating things like the same slot schedule, connections, topology, etc. over multiple runs is super useful.